### PR TITLE
Add tooltip next to search bar

### PR DESCRIPTION
**Pull request by sebwittr**
# Purpose

###### A brief description of the purpose of the PR's changes for the project.

<br> Adds a tooltip next to the search bar, in order to show the user how to use phrase searches. 

# Tickets

###### A link to any tickets this PR is associated with on Trello.

- https://trello.com/c/LVcD61Yq/318-tooltip-for-phrase-searching

# Contributors

###### Anyone who contributed to this PR for future reference.

- @sebwittr 

# Feature List

###### A list of more in-depth and technical changes, additions, deletions, and fixes this PR provides.

- Adds a SearchInfoIcon component that contains the icon, along with the tooltip itself.
-
-

# Notes

###### A list of miscellaneous notes for this pull request, such as whether we need to add something later for production, or that it was time boxed, etc.

<br>

# Checklist

- [x] Filled out PR template :wink:
- [x] Approved by designers
- [x] Is passing linting checks
- [x] Is passing tsc
- [x] Is passing existing tests
- [x] Has documentation and comments in code
- [x] Has test coverage for code
- [x] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

**Comment by hankewyczz** [comment]
image

**Review by Lucas-Dunker — APPROVED** [review]
Code is intuitive to understand, looks fine to me; the tooltip is laid under the husky icon when in smaller window sizes, but that seems to be an overall issue with screen sizing not limited to just the tooltip.

**Comment by sebwittr** [comment]
> image

This should be fixed with the newest commit!

File: styles/home/_HomeSearch.scss [hunk 1]
```diff
@@ -86,6 +85,7 @@
       border-radius: 0 5px 5px 0;
       height: 46px;
       width: 100%;
+      display: flex;
 
       & > div {
```
Comment (koulopoulos) [inline styles/home/_HomeSearch.scss:90]: It looks like this selector `.HomeSearch__searchBar--input > div` is causing the new icon to have a faint white border to the left (see attached image). This looks a little awkward and should probably be removed.

Screen Shot 2022-10-13 at 3 48 35 PM

**Review by koulopoulos — CHANGES_REQUESTED** [review]
The `z-index` fix looks good 👍 - Just one small comment on the styling.

File: styles/home/_HomeSearch.scss (hunk 1 above)
Comment (sebwittr) [reply styles/home/_HomeSearch.scss:90]: This should be fixed now

File: styles/home/_HomeSearch.scss [hunk 2]
```diff
@@ -86,14 +85,14 @@
...
         height: 100%;
         width: 100%;
         background: #f5f5f5;
         border-top: none;
         border-bottom: none;
-        border-left: 1px solid #e8e8e8;
```
Comment (koulopoulos) [inline styles/home/_HomeSearch.scss:96]: Looks like the removal of this style also effected the border between the semester dropdown and text input. Not a huge deal, as it's just a difference in the shade of gray, but noting it here.

Screen Shot 2022-10-16 at 2 52 31 PM

**Review by koulopoulos — APPROVED** [review]
Just noting a style change, but nothing major.

File: components/ResultsPage/SearchBar.tsx [hunk 3]
```diff
@@ -44,43 +44,48 @@ export default function SearchBar({
...
+          style={{ background: buttonColor }}
+        >
+          <MagnifyingGlass
+            aria-label="magnifying glass"
+            className="searchbar__magnifyingGlass"
+          />
+        </div>
       </div>
-    </div>
+      <SearchInfoIcon />
```
Comment (hankewyczz) [inline components/ResultsPage/SearchBar.tsx]: Currently, the icon doesn't do anything on mobile. We can do either one of two things:
- Remove it for mobile views
- Have it display the tooltip on click, not just on hover

Also, the home screen has this border on mobile 
image

**Review by hankewyczz — CHANGES_REQUESTED** [review]

File: components/ResultsPage/SearchBar.tsx (hunk 3 above)
Comment (sebwittr) [reply components/ResultsPage/SearchBar.tsx]: I got rid of the tooltip for mobile, so these problems shouldn't exist anymore

File: components/ResultsPage/SearchBar.tsx [hunk 4]
```diff
@@ -44,43 +44,48 @@ export default function SearchBar({
...
+          style={{ background: buttonColor }}
+        >
+          <MagnifyingGlass
+            aria-label="magnifying glass"
+            className="searchbar__magnifyingGlass"
+          />
+        </div>
       </div>
-    </div>
+      {!macros.isMobile ? <SearchInfoIcon /> : <></>}
```
Comment (hankewyczz) [inline components/ResultsPage/SearchBar.tsx:88]: In the rest of our code, we tend to use `{!macros.isMobile && <SearchInfoIcon />`

(This is just for context to see what's equivalent - I think both ways are confusing and poorly-written (thanks react) so i'm fine with you using either one here)

**Review by hankewyczz — APPROVED** [review]

**Pull request merged**